### PR TITLE
Add QEMU support for NuttX service (RDT-1034)

### PR DESCRIPTION
**Pull request by fdcavalcanti**
## Description
This PR improves the Nuttx service by adding support for QEMU.
It also modifies the reset functionality to support serial reset (DTR toggle), QEMU reset and ESP reset.
On a different topic, there's a smaller commit to improve logging on app binary search.

## Related
None.

## Testing
Tested locally running qemu applications.

---

Looking forward to any improvements.

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [ ] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

File: pytest-embedded-nuttx/pytest_embedded_nuttx/__init__.py [hunk 1]
```diff
@@ -4,20 +4,23 @@
 
 from pytest_embedded.utils import lazy_load
 
-from .dut import NuttxDut, NuttxEspDut
+from .app import NuttxApp
```
Comment (hfudev) [inline pytest-embedded-nuttx/pytest_embedded_nuttx/__init__.py:7]: `.app` requires `esptool`, which is an optional dependency. so better to keep it in the lazy-load part.

File: pytest-embedded-nuttx/pytest_embedded_nuttx/dut.py [hunk 2]
```diff
@@ -75,8 +75,78 @@ def write_and_return(self, data: str, timeout: int = 2) -> AnyStr:
...
+
+    def reset(self) -> None:
+        """Reset the DUT by toggling the DTR line.
+
+        Args:
+            None.
+
+        Returns:
+            None.
+        """
```
Comment (hfudev) [inline pytest-embedded-nuttx/pytest_embedded_nuttx/dut.py:117]:
```suggestion
        """Reset the DUT by toggling the DTR line."""
```

to simplify it. (a few more occurrences)

**Review by hfudev — COMMENTED** [review]
LGTM! Thanks for the PR.

Besides, could you also update the optional dependency `qemu` here?

https://github.com/espressif/pytest-embedded/blob/8b811667b186ff492c4eb19f133ac7954d5ee641/pytest-embedded-nuttx/pyproject.toml#L38

and here with the optional `qemu` service?

https://github.com/espressif/pytest-embedded/blob/b4c564ba2c584a96d89a607be3b258fdb78a1b52/docs/concepts/services.md?plain=1#L20

File: pytest-embedded-nuttx/pytest_embedded_nuttx/__init__.py (hunk 1 above)
Comment (fdcavalcanti) [reply pytest-embedded-nuttx/pytest_embedded_nuttx/__init__.py:7]: Sorry I don't follow what modification is supposed to go here. That line 7 should go away entirely?

Comment (fdcavalcanti) [reply pytest-embedded-nuttx/pytest_embedded_nuttx/__init__.py]: Thanks!

**Comment by fdcavalcanti** [comment]
@hfudev Modifications done. Also added few changes on documentation regarding qemu.

**Comment by hfudev** [comment]
LGTM. Thank you!

**Pull request merged**